Tribler/Core/Modules/restapi/channels/channels_playlists_endpoint.py

```diff
@@ -227,3 +230,119 @@ def render_POST(self, request):
                                                        'description': parameters['description'][0]})
 
         return json.dumps({"modified": True})
+
+
+class ChannelsModifyPlaylistTorrentsEndpoint(BaseChannelsEndpoint):
+
+    def __init__(self, session, cid, playlist_id, infohash):
+        BaseChannelsEndpoint.__init__(self, session)
+        self.cid = cid
+        self.playlist_id = playlist_id
+        self.infohash = infohash.decode('hex')
+
+    def render_PUT(self, request):
+        """
+        .. http:put:: /channels/discovered/(string: channelid)/playlists/(int: playlistid)/(string: infohash)
+
+        Add a torrent with a specified infohash to a specified playlist. The torrent that is added to the playlist,
+        should be present in the channel.
+
+            **Example request**:
+
+            .. sourcecode:: none
+
+                curl -X PUT http://localhost:8085/channels/discovered/abcd/playlists/3/abcdef
+
+            **Example response**:
+
+            .. sourcecode:: javascript
+
+                {
+                    "added": True
+                }
+
+            :statuscode 404: if the specified channel/playlist/torrent does not exist.
+            :statuscode 409: if the specified torrent is already in the specified playlist.
+        """
+        channel_info = self.get_channel_from_db(self.cid)
+        if channel_info is None:
+            return ChannelsPlaylistsEndpoint.return_404(request)
+
+        playlist = self.channel_db_handler.getPlaylist(self.playlist_id, ['Playlists.dispersy_id'])
+        if playlist is None:
+            return BaseChannelsEndpoint.return_404(request, message="this playlist cannot be found")
+
+        channel_community = self.get_community_for_channel_id(channel_info[0])
```

Comment:
Good point, I was able to move this statement to another location, possibly saving a database query.